contracts/facets/GBMFacet.sol

```diff
@@ -297,13 +368,106 @@ contract GBMFacet is IGBM, IERC1155TokenReceiver, IERC721TokenReceiver, Modifier
             s.auctions[_auctionId].bidMultiplier = s.initiatorInfo.bidMultiplier;
         }
 
+        if (_auctionPresetID != 0) {
+            s.auctions[_auctionId].owner = LibDiamond.contractOwner();
+            s.auctions[_auctionId].startTime = s.auctionPresets[_auctionPresetID].startTime;
+            s.auctions[_auctionId].endTime = s.auctionPresets[_auctionPresetID].endTime;
+            s.auctions[_auctionId].hammerTimeDuration = s.auctionPresets[_auctionPresetID].hammerTimeDuration;
+            s.auctions[_auctionId].bidDecimals = s.auctionPresets[_auctionPresetID].bidDecimals;
+            s.auctions[_auctionId].stepMin = s.auctionPresets[_auctionPresetID].stepMin;
+            s.auctions[_auctionId].incMin = s.auctionPresets[_auctionPresetID].incMin;
+            s.auctions[_auctionId].incMax = s.auctionPresets[_auctionPresetID].incMax;
+            s.auctions[_auctionId].bidMultiplier = s.auctionPresets[_auctionPresetID].bidMultiplier;
+
+            require(s.auctions[_auctionId].startTime != 0, "Auction startTime has not been set");
+        }
+
         //Event emitted when an auction is being setup
-        emit Auction_Initialized(_auctionId, _tokenId, _1155Index, _tokenContract, _tokenKind);
+        emit Auction_Initialized(_auctionId, _tokenId, _locIndex, _tokenContract, _tokenKind);
 
         //Event emitted when the start time of an auction changes (due to admin interaction )
         emit Auction_StartTimeUpdated(_auctionId, getAuctionStartTime(_auctionId));
     }
 
+	/// @notice Seller can cancel an auction during the grace period 
+    /// Throw if the token owner is not the caller of the function
+    /// @param _auctionID The auctionId of the auction to complete
+    function cancelAuction(uint256 _auctionID) public {
+		require(s.auctionSeller[_auctionID] != address(0), "Initial auctions cannot be cancelled");
+		
+        address _ca = s.tokenMapping[_auctionID].contractAddress;
+        uint256 _tid = s.tokenMapping[_auctionID].tokenId;
+        uint256 _tam = s.tokenMapping[_auctionID].tokenAmount;
+        
+        require(s.collections[_ca].biddingAllowed, "claim: Claiming is currently not allowed");
+        require(getAuctionEndTime(_auctionID) < block.timestamp, "cancelAuction: Auction has not yet ended");
+        require(getAuctionEndTime(_auctionID) + getAuctionHammerTimeDuration(_auctionID) > block.timestamp, "cancelAuction: Cancellation time has already ended");
+        
+        require(!s.auctionItemClaimed[_auctionID], "claim: this auction has alredy been claimed");   
+        s.auctionItemClaimed[_auctionID]  = true;
+
+		uint256 _proceeds = s.auctions[_auctionID].highestBid - s.auctions[_auctionID].auctionDebt;
+		
+		//Send the debt + his due incentives from the seller to the highest bidder
```

Comment:
//to-do
The seller should only cover the incentives.
The GBM contract covers the debt

Reply:
Also, i don't see anywhere where the owner of an auction gets his reward after an auction has been completed